phoenix-core/src/main/java/org/apache/phoenix/util/ByteUtil.java

```diff
@@ -508,6 +508,26 @@ public static byte[] fillKey(byte[] key, int length) {
         return newBound;
     }
 
+    public static boolean isEmptyOrNull(byte[] b, int offset, int length) {
+        if (b == null) {
+            return true;
+        }
+        if (Bytes.compareTo(b, HConstants.EMPTY_BYTE_ARRAY) == 0 ) {
+            return true;
+        }
+        if (b.length < offset + length) {
+            throw new IllegalStateException();
+        }
+        boolean result = true;
```

Comment:
I think we can rewrite to something like this?
`        
        for (int i = (offset + length) - 1; i >= offset; i--) {
            if (b[i] != 0x0) {
                return false;
            }
        }
        return true;`